rewriter/Minitest.cc

```diff
@@ -653,6 +654,99 @@ ast::ExpressionPtr runSingle(core::MutableContext ctx, bool isClass, ast::Send *
+                return nullptr;
+            }
+
+            auto &arg = send->getPosArg(0);
+
+            // Only handle symbol arguments for now (its(:attribute))
+            auto argLit = ast::cast_tree<ast::Literal>(arg);
+            if (argLit == nullptr || !argLit->isName()) {
+                return nullptr;
+            }
+
+            auto attributeName = argLit->asName();
+            auto argString = attributeName.show(ctx);
+
+            // Create the describe block name
+            auto describeTestName = fmt::format("<describe '{}'>", argString);
+            auto describeName =
+                ast::MK::UnresolvedConstantParts(arg.loc(), {ctx.state.enterNameConstant(describeTestName)});
+
+            auto isExpectedName = ctx.state.enterNameUTF8("is_expected");
+            auto expectName = ctx.state.enterNameUTF8("expect");
+
+            // Transform the its block body to replace is_expected with expect(subject.attribute)
+            class IsExpectedTransformer {
+            private:
+                core::NameRef attributeName;
+                core::NameRef isExpectedName;
+                core::NameRef expectName;
+
+            public:
+                IsExpectedTransformer(core::NameRef attributeName, core::NameRef isExpectedName,
+                                      core::NameRef expectName)
+                    : attributeName(attributeName), isExpectedName(isExpectedName), expectName(expectName) {}
+
+                ast::ExpressionPtr postTransformSend(core::Context ctx, ast::ExpressionPtr tree) {
+                    auto send = ast::cast_tree<ast::Send>(tree);
+                    if (send == nullptr) {
+                        return tree;
+                    }
+
+                    // Look for is_expected calls
+                    if (send->fun == isExpectedName && send->recv.isSelfReference()) {
+                        // Replace is_expected with expect(subject.attribute)
+                        auto subjectCall = ast::MK::Send0(send->loc, ast::MK::Self(send->loc), core::Names::subject(),
+                                                          send->loc.copyWithZeroLength());
+                        auto attributeCall = ast::MK::Send0(send->loc, std::move(subjectCall), attributeName,
+                                                            send->loc.copyWithZeroLength());
+                        return ast::MK::Send1(send->loc, ast::MK::Self(send->loc), expectName,
+                                              send->loc.copyWithZeroLength(), std::move(attributeCall));
+                    }
+
+                    return tree;
+                }
+            };
+
+            ast::ExpressionPtr itBody = std::move(block->body);
+            IsExpectedTransformer transformer(attributeName, isExpectedName, expectName);
+            itBody = ast::TreeMap::apply(ctx, transformer, std::move(itBody));
+
+            // Create it block
+            auto itName = ctx.state.enterNameUTF8("<it>");
```

Comment:
does this make sense as an anonymous it? it will always be in the context of a non-anonymous `describe` so it seemed ok

Reply:
I don't have an opinion whether it's anonymous or not, but we should never have statically known names entered anywhere other than `generate_names.cc`.

Reply:
76a6ac112